src/utils.ts

```diff
@@ -98,14 +101,92 @@ export const extractEvents = (
 
     switch (decodedEvent.name) {
       case InvariantEventNames.CreatePositionEvent:
-        eventsObject[InvariantEventNames.CreatePositionEvent].push(
-          parseEvent(decodedEvent)
-        );
+        const parsedCreateEvent: CreatePositionEvent = parseEvent(decodedEvent);
```

Comment:
Some of the fields are used in many places, it will enhance the readability if you extract the values from object.
For example instead casting multiple times owner to string you can introduce
```ts
const {owner} = parsedCreateEvent
const ownerKey = owner.toString()
```

Apply it to all similiar places please